pyflakes/checker.py

```diff
@@ -1699,7 +1704,33 @@ def CALL(self, node):
                     with self._enter_annotation():
                         self.handleNode(keyword.value, node)
 
-        self.handleChildren(node)
+        if _is_typing(node.func, "TypedDict", self.scopeStack):
+            # TypedDict("a", {"a": int})
+            if len(node.args) > 1 and isinstance(node.args[1], ast.Dict):
+                self._annotated_nodes += node.args[1].values
+
+            # TypedDict("a", a=int)
+            self._annotated_nodes += [k.value for k in node.keywords]
```

Comment:
🤦 just use `self._enter_annotation` inline right here, don't introduce state that's unneeded -- look at how `TypeVar` right above this works

Reply:
I can't only use `self._enter_annotation`, I need some info that some nodes are potential types while entering with `AnnotationState.NONE` to prevent flagging all other STR as forward references.
Your facepalm is really uncalled for

Reply:
you have that information!  it's the same as the things you're assinging to `self._annotated_nodes`

Reply:
I know I have that information, but I need to handle nested children nodes differently by their "annotated" property or not. I don't think removing these nodes from the tree would be good practice. How would you make it so they're analyzed just once ?

Reply:
follow how `TypeVar` does it?

Reply:
Following TypeVar implementation doesn't solve this case
```python
from typing import NamedTuple
from t import A
from t import C

NamedTuple("A", [("a", A["C"])])
```

Reply:
TypeVar in this case does not support it either
```python
from typing import TypeVar
from t import A
from t import C

TypeVar("A", bound=A["C"])
```
```
t.py:3:1 't.C' imported but unused
```
Even though for TypeVar this is an incorrect syntax anyway so it doesn't matter

Reply:
I see, looks like TypeVar has a slight bug:

```diff
diff --git a/pyflakes/checker.py b/pyflakes/checker.py
index 57603c8..1d18a1e 100644
--- a/pyflakes/checker.py
+++ b/pyflakes/checker.py
@@ -1689,13 +1689,12 @@ class Checker(object):
         elif _is_typing(node.func, 'TypeVar', self.scopeStack):
             # TypeVar("T", "int", "str")
             for arg in node.args[1:]:
-                if isinstance(arg, ast.Str):
-                    with self._enter_annotation():
-                        self.handleNode(arg, node)
+                with self._enter_annotation():
+                    self.handleNode(arg, node)
 
             # TypeVar("T", bound="str")
             for keyword in node.keywords:
-                if keyword.arg == 'bound' and isinstance(keyword.value, ast.Str):
+                if keyword.arg == 'bound':
                     with self._enter_annotation():
                         self.handleNode(keyword.value, node)
 
```

Reply:
#610 for that (if you want to review it)

Reply:
You can't do that because now you're allowing some nodes to be analyzed 2 times. With your implementation,
```python
from typing import TypeVar, List

TypeVar("A", bound=List["A"])
```

```
t.py:3:25 undefined name 'A'
t.py:3:25 undefined name 'A'
```

Reply:
that's existing it seems -- not hard to pick apart the `Call` though and visit each member individually

Reply:
That's what I wanted to do at first, what would be the best way to do that without almost recreating a `self.handleNode` ?